src/VisualStudio/Core/Test/Progression/ProgressionTestHelpers.vb

```diff
@@ -36,10 +36,19 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Progression
 
         Public Sub AssertSimplifiedGraphIs(graph As Graph, xml As XElement)
             Dim graphXml = graph.ToSimplifiedXDocument()
-            If Not XNode.DeepEquals(graphXml, xml) Then
+            If Not XNode.DeepEquals(graphXml.Root, xml) Then
                 ' They aren't equal, so therefore the text representations definitely aren't equal.
                 ' We'll Assert.Equal those, so that way xunit will show nice before/after text
-                Assert.Equal(xml.ToString(), graphXml.ToString())
+                'Assert.Equal(xml.ToString(), graphXml.ToString())
+
+                ' In an attempt to diagnose some flaky tests, the whole contents of both objects will be output
+                Throw New Exception($"Graph XML was not equal, check for out-of-order elements.
```

Comment:
Because the current output (taken from the bug) doesn't contain enough info, it's just:

```
Assert.Equal() Failure
                                 ↓ (pos 101)
Expected: ··· <Node Id="(@1 Type=A)" Category="CodeSchema_Class" CodeSchem···
Actual:   ··· <Node Id="(@1 Type=B)" Category="CodeSchema_Class" CodeSchem···
                                 ↑ (pos 101)
```

And my current theory is that the resultant XML is 'correct', just out-of-order, and if that's the case then the end of this XML fragment would have a reversed ordering of the test assemblies and that's ultimately what I'm trying to see.